iib/workers/tasks/build.py

```diff
@@ -840,6 +908,39 @@ def handle_add_request(
             (prebuild_info['distribution_scope'] in ['dev', 'stage']),
         )
 
+        # Add the max ocp version property
+        # We need to ensure that any bundle which has deprecated/removed API(s) in 1.22/ocp 4.9
+        # will have this property to prevent users from upgrading clusters to 4.9 before upgrading
+        # the operator installed to a version that is compatible with 4.9
+
+        # Get the CSV name and version (not just the bundle path)
+        db_path = temp_dir + "/database/index.db"
+        port, rpc_proc = _serve_index_registry(db_path)
+
+        rawBundles = run_cmd(
+            ['grpcurl', '-plaintext', f'localhost:{port}', 'api.Registry/ListBundles'],
+            exc_msg='Failed to get bundle data from index image',
+        )
+        rpc_proc.kill()
+
+        # Get bundle json for bundles in the request
+        updated_bundles = list(
+            filter(lambda b: b['bundlePath'] in resolved_bundles, _get_bundle_json(rawBundles))
+        )
+
+        for bundle in updated_bundles:
+            if _requires_max_ocp_version(bundle['bundlePath']):
+                log.info('adding property for %s', bundle['bundlePath'])
+                max_openshift_version_property = {
+                    'type': 'olm.maxOpenShiftVersion',
+                    'value': '4.8',
```

Comment:
can this be configurable? Or rather, can this be changed in the future? If yes, lets make it configurable

Reply:
There is no other valid configuration today than "4.8". If you are using APIs that will be removed in 4.9 then the max ocp version should be set to "4.8" and no other value. If that changes for some reason we should track it as a bug at that time IMO